From 5fe707c2415dfbf917a2e918c0c7238e2f159df8 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd.simons@collabora.co.uk>
Date: Mon, 1 Mar 2010 13:58:58 +0000
Subject: [PATCH] Only time out channel requests after many hours

Reviewed-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
---
 src/mcd-connection.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/mcd-connection.c b/src/mcd-connection.c
index 72d2bd3..8cfe1bd 100644
--- a/src/mcd-connection.c
+++ b/src/mcd-connection.c
@@ -2363,14 +2363,16 @@ request_channel_new_iface (McdConnection *connection, McdChannel *channel)
     properties = _mcd_channel_get_requested_properties (channel);
     if (_mcd_channel_get_request_use_existing (channel))
     {
+        /* Timeout of 5 hours: 5 * 3600 * 1000 */
         tp_cli_connection_interface_requests_call_ensure_channel
-            (priv->tp_conn, -1, properties, ensure_channel_cb,
+            (priv->tp_conn, 18000000, properties, ensure_channel_cb,
              connection, NULL, (GObject *)channel);
     }
     else
     {
+        /* Timeout of 5 hours: 5 * 3600 * 1000 */
         tp_cli_connection_interface_requests_call_create_channel
-            (priv->tp_conn, -1, properties, create_channel_cb,
+            (priv->tp_conn, 18000000, properties, create_channel_cb,
              connection, NULL, (GObject *)channel);
     }
     return TRUE;
