From cc746edd95fdc1802cd656eca664a7089de61745 Mon Sep 17 00:00:00 2001
From: Jonny Lamb <jonny.lamb@collabora.co.uk>
Date: Fri, 07 Oct 2011 10:34:52 +0000
Subject: connectivity-monitor: fix build with UPower but no NM or ConnMan

http://bugs.debian.org/644589

Signed-off-by: Jonny Lamb <jonny.lamb@collabora.co.uk>
---
diff --git a/src/connectivity-monitor.c b/src/connectivity-monitor.c
index 4ecccde..2c94e3f 100644
--- a/src/connectivity-monitor.c
+++ b/src/connectivity-monitor.c
@@ -317,10 +317,12 @@ mcd_connectivity_monitor_init (McdConnectivityMonitor *connectivity_monitor)
 static void
 connectivity_monitor_finalize (GObject *object)
 {
-#ifdef HAVE_NM
+#if defined(HAVE_NM) || defined(HAVE_CONNMAN) || defined(HAVE_UPOWER)
   McdConnectivityMonitor *connectivity_monitor = MCD_CONNECTIVITY_MONITOR (object);
   McdConnectivityMonitorPrivate *priv = connectivity_monitor->priv;
+#endif
 
+#ifdef HAVE_NM
   if (priv->nm_client != NULL)
     {
       g_signal_handler_disconnect (priv->nm_client,
@@ -332,9 +334,6 @@ connectivity_monitor_finalize (GObject *object)
 #endif
 
 #ifdef HAVE_CONNMAN
-  McdConnectivityMonitor *connectivity_monitor = MCD_CONNECTIVITY_MONITOR (object);
-  McdConnectivityMonitorPrivate *priv = connectivity_monitor->priv;
-
   if (priv->proxy != NULL)
     {
       dbus_g_proxy_disconnect_signal (priv->proxy, "StateChanged",
--
cgit v0.9.0.2-2-gbebe
